File: solutions/javascript/bob/4/bob.js
export const hey = (message) => {
  let mes = message.trim()

  if (!mes) return "Fine. Be that way!"

  if (mes.toLowerCase() !== mes && mes.toUpperCase() === mes){
    return (mes.endsWith("?")) ? "Calm down, I know what I'm doing!" : "Whoa, chill out!"
  }

  return (mes.endsWith("?")) ? "Sure." : "Whatever."

};

File: solutions/python/bob/4/bob.py
def response(hey_bob):
    hey_bob = hey_bob.strip()
    # isupper() return True if all cased letters are uppercase, False if there are **no letters** or have lowercase letters.
    if hey_bob.isupper():
        if hey_bob.endswith("?"):
            return "Calm down, I know what I'm doing!"
        return "Whoa, chill out!"
    elif hey_bob.endswith("?"):
        return "Sure."
    elif hey_bob == "":
        return "Fine. Be that way!"
    return "Whatever."
